homeassistant/components/bluetooth/base_scanner.py

```diff
@@ -28,10 +30,26 @@
 class BaseHaScanner:
     """Base class for Ha Scanners."""
 
-    def __init__(self, hass: HomeAssistant, source: str) -> None:
+    __slots__ = ("hass", "source", "_connecting", "name", "scanning")
+
+    def __init__(self, hass: HomeAssistant, source: str, adapter: str) -> None:
         """Initialize the scanner."""
         self.hass = hass
         self.source = source
+        self._connecting = 0
+        self.name = adapter_human_name(adapter, source) if adapter != source else source
+        self.scanning = True
+
+    @contextmanager
+    def connecting(self) -> Generator[None, None, None]:
+        """Context manager to track connecting state."""
+        self._connecting += 1
+        self.scanning = not self._connecting
```

Comment:
I originally did but the property can get accessed 25000 times per minute since it's in the advertisement hot path so I changed it to an attribute so it's just a dict lookup under the hood.  Can't wait for python 3.11.